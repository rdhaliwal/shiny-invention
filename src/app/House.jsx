import React from 'react';

export default class House extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className={`House House--${this.props.name.toLowerCase()}`}>
        <h2 className="House-name">
          {this.props.name}
        </h2>
        <div className="House-logo"></div>
      </div>
    );
  }
}

House.defaultProps = {
  name: "No House"
}
