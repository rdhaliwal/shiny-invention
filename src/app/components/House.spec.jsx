// Link.react-test.js
import React from 'react';
import House from './House';
import renderer from 'react-test-renderer';

test('It renders the Gryffindor House Component', () => {
  let props =  {
    match : {params:{houseName: "Gryffindor"}}
  }
  const component = renderer.create(
    <House {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('It renders the default House Component when no props are given', () => {
  const component = renderer.create(
    <House />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
