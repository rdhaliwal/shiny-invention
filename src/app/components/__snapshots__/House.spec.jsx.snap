// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders the Gryffindor House Component 1`] = `
<div
  className="House House--Gryffindor"
>
  <h2
    className="House-name"
  >
    Gryffindor
  </h2>
  <div
    className="House-logo"
  />
</div>
`;

exports[`It renders the default House Component when no props are given 1`] = `
<div
  className="House House--No House"
>
  <h2
    className="House-name"
  >
    No House
  </h2>
  <div
    className="House-logo"
  />
</div>
`;
